plugin-cgroupv1/src/oar2/plugin.rs

```diff
@@ -134,131 +100,29 @@ impl AlumetPlugin for Oar2Plugin {
-                                    cpu_metric: job_detect.cpu_metric,
-                                    memory_metric: job_detect.memory_metric,
-                                    oar2_metric_file: metric_file,
-                                });
-
-                                let source_name = job_name;
-                                let trigger = TriggerSpec::at_interval(job_detect.poll_interval);
-                                let create_source = request::create_one().add_source(source_name, new_source, trigger);
-
-                                job_detect
-                                    .rt
-                                    .block_on(
-                                        job_detect
-                                            .control_handle
-                                            .dispatch(create_source, Duration::from_millis(500)),
-                                    )
-                                    .with_context(|| format!("failed to add source {source_name}"))?;
-                            }
-                        }
-                    }
-                    Ok(())
-                }
-
-                log::debug!("Handle event function");
-                if let Ok(Event {
-                    kind: EventKind::Create(_),
-                    paths,
-                    ..
-                }) = event
-                {
-                    log::debug!("Paths: {paths:?}");
-                    for path in paths {
-                        if let Err(e) = handle_event_on_path(self, path.clone()) {
-                            log::error!("Unable to handle event on {}: {}", path.display(), e);
-                        }
-                    }
-                } else if let Err(e) = event {
-                    log::error!("watch error: {:?}", e);
-                }
-            }
-        }
+        let trigger = self.trigger.clone();
+        let metrics = self.metrics.as_ref().unwrap().clone();
+        let cpuacct_controller_path = self.cpuacct_controller_path.clone();
+        let memory_controller_path = self.memory_controller_path.clone();
 
         let rt = tokio::runtime::Builder::new_current_thread()
             .enable_time()
             .build()
             .context("tokio Runtime should build")?;
 
         let handler = JobDetector {
-            config_path: config_path.clone(),
-            cpu_metric,
-            memory_metric,
+            cpuacct_controller_path: cpuacct_controller_path.clone(),
+            memory_controller_path: memory_controller_path.clone(),
             control_handle,
-            poll_interval,
+            metrics,
+            trigger,
             rt,
         };
         let mut watcher = notify::recommended_watcher(handler)?;
 
-        watcher.watch(&config_path.join("cpuacct/oar"), RecursiveMode::NonRecursive)?;
-        watcher.watch(&config_path.join("memory/oar"), RecursiveMode::NonRecursive)?;
+        println!("START WATCHER");
```

Comment:
a little println is lost here, please bring it back to his parents and use `log` instead :)